### PR TITLE
PL-1004 re-added docker

### PR DESCRIPTION
**Pull request by ags313**

### CODE REVIEW

File: baseci/Dockerfile [hunk 1]
```diff
@@ -64,11 +65,6 @@ RUN mkdir $HOME/.gnupg && \
     apt-get update && \
     apt-get -y --no-install-recommends install mono-complete
 
-# install latest git
```
Comment (christopherschwaab) [inline baseci/Dockerfile:67]: Are we sure the version of git is okay?  I recall the version supplied for focal is too old.

**Review by christopherschwaab — CHANGES_REQUESTED** [review]

**Comment by christopherschwaab** [comment]
I think 2.32 or 2.33 (we need sparse index (cone mode)

File: baseci/Dockerfile (hunk 1 above)
Comment (ags313) [reply baseci/Dockerfile:67]: 2.25 for 20.04 according to https://packages.ubuntu.com/search?keywords=git

**Comment by kwahsog** [comment]
```
 => ERROR [13/36] RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/  0.3s
------
 > [13/36] RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg:
#17 0.204 gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
#17 0.205 gpg: can't create '/etc/apt/keyrings/docker.gpg': No such file or directory
#17 0.205 gpg: no valid OpenPGP data found.
#17 0.205 gpg: dearmoring failed: No such file or directory
#17 0.253 (23) Failed writing body
------
executor failed running [/bin/sh -c curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg]: exit code: 2
alex@GS-MacBook-Pro baseci % code .
alex@GS-MacBook-Pro baseci % docker build .
[+] Building 3.2s (16/37)
 => [internal] load build definition from Dockerfile                                                        0.0s
 => => transferring dockerfile: 5.44kB                                                                      0.0s
 => [internal] load .dockerignore                                                                           0.0s
 => => transferring context: 2B                                                                             0.0s
 => [internal] load metadata for docker.io/library/ubuntu:20.04                                             0.5s
 => [ 1/34] FROM docker.io/library/ubuntu:20.04@sha256:450e066588f42ebe1551f3b1a535034b6aa46cd936fe7f2c6b0  0.0s
 => CACHED [ 2/34] RUN apt-get update &&     apt-get upgrade -y &&     apt-get install -y --no-install-rec  0.0s
 => CACHED [ 3/34] RUN apt-get install -y gnupg2                                                            0.0s
 => CACHED [ 4/34] RUN wget -qO - https://apt.corretto.aws/corretto.key | apt-key add -                     0.0s
 => CACHED [ 5/34] RUN add-apt-repository --yes 'deb https://apt.corretto.aws stable main' &&     apt-get   0.0s
 => CACHED [ 6/34] RUN wget -q https://dl.google.com/go/go1.18.linux-amd64.tar.gz &&     tar -xzf go1.18.l  0.0s
 => CACHED [ 7/34] RUN wget http://archive.apache.org/dist/maven/maven-3/3.8.6/binaries/apache-maven-3.8.6  0.0s
 => CACHED [ 8/34] RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v3.10.0/protoc-3  0.0s
 => CACHED [ 9/34] RUN mkdir $HOME/.gnupg &&     echo "disable-ipv6" >> $HOME/.gnupg/dirmngr.conf &&     a  0.0s
 => CACHED [10/34] RUN apt-get install -y --no-install-recommends curl                                      0.0s
 => CACHED [11/34] RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.20.0/bin/linu  0.0s
 => CACHED [12/34] RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" &  0.0s
 => ERROR [13/34] RUN add-apt-repository    "deb [arch=amd64] https://download.docker.com/linux/ubuntu      2.5s
------
 > [13/34] RUN add-apt-repository    "deb [arch=amd64] https://download.docker.com/linux/ubuntu    $(lsb_release -cs)    stable":
#16 0.853 Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
#16 0.853 Hit:2 http://archive.ubuntu.com/ubuntu focal-updates InRelease
#16 0.853 Hit:3 http://archive.ubuntu.com/ubuntu focal-backports InRelease
#16 0.853 Hit:4 http://security.ubuntu.com/ubuntu focal-security InRelease
#16 0.867 Hit:5 https://download.mono-project.com/repo/ubuntu stable-bionic InRelease
#16 0.870 Get:6 https://download.docker.com/linux/ubuntu focal InRelease [57.7 kB]
#16 0.871 Hit:7 https://apt.corretto.aws stable InRelease
#16 1.598 Err:6 https://download.docker.com/linux/ubuntu focal InRelease
#16 1.598   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7EA0A9C3F273FCD8
#16 1.612 Reading package lists...
#16 2.513 W: GPG error: https://download.docker.com/linux/ubuntu focal InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7EA0A9C3F273FCD8
#16 2.513 E: The repository 'https://download.docker.com/linux/ubuntu focal InRelease' is not signed.
------
executor failed running [/bin/sh -c add-apt-repository    "deb [arch=amd64] https://download.docker.com/linux/ubuntu    $(lsb_release -cs)    stable"]: exit code: 100

**Pull request closed without merging**